### PR TITLE
Cleanup book build

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 some leftovers in gitbooking our book has build artifacts checked into
 the source repository

 #### Summary of Changes
 * remove the build artifacts (they're built anyway)
 * add .gitignore for the intermediate files
 * whack the build-svg script
 * move updating gitbook-cage into solana-secondary (publish-book.sh)

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
For this to work, we need to make changes to https://buildkite.com/solana-labs/gitbook-mirror-sync/settings
`ci/publish-book.sh` publishes to the "old" book location but not gitbook (the "new" location).  I haven't fully purged the "old" book location because I'm still not 100% on gitbook -- I like how it looks more but not sure I fully trust it yet

**Comment by rob-solana** [comment]
> For this to work, we need to make changes to https://buildkite.com/solana-labs/gitbook-mirror-sync/settings
> `ci/publish-book.sh` publishes to the "old" book location but not gitbook (the "new" location). I haven't fully purged the "old" book location because I'm still not 100% on gitbook -- I like how it looks more but not sure I fully trust it yet

the cage stuff doesn't run our makefile, it just runs mdbook?

**Comment by mvines** [comment]
No gitbook imports the markdown directly from the cage github project and renders on their end, mdbook is unused.

The export to cage would need to create the svgs for gitbook.

**Comment by rob-solana** [comment]
> No gitbook imports the markdown directly from the cage github project and renders on their end, mdbook is unused.
> 
> The export to cage would need to create the svgs for gitbook.

means running 'make' in the solanalabs/rust docker, right?

**Comment by mvines** [comment]
1. Run `make -j"$(nproc)" -B svg`
2. Commit the resulting svgs

**Comment by rob-solana** [comment]
> 1. Run `make -j"$(nproc)" -B svg`
> 2. Commit the resulting svgs

yeah, but the svgs are from svgbob, in the docker image

**Comment by garious** [comment]
Any news here?

**Comment by rob-solana** [comment]
looking at it again now

we have to clone non-raw

**Comment by rob-solana** [comment]
@garious why did you update the SVGs to have -NN.svg names?

**Comment by garious** [comment]
That doesn't sound like me. What are you looking at exactly?

**Comment by rob-solana** [comment]
maybe wasn't you..  9fbbb17c3b

**Comment by garious** [comment]
@ryan-shea

**Comment by rob-solana** [comment]
I think the job's updated to support this, @mvines have a look:
https://buildkite.com/solana-labs/gitbook-mirror-sync/builds/1533#af67f54a-47fe-460c-83c7-dec5603af16b

**Comment by mvines** [comment]
As we discussed IRL, seems like nice path forward is to move the gitbook-mirror-sync pipeline into `ci/publish-book.sh`

**Comment by ryan-shea** [comment]
@garious @rob-solana didn't do anything on the gitbook side. let me know how I can help; some of the build stuff w/ docker is a tad over my head atm.

**Comment by rob-solana** [comment]
@mvines ready for another look

I tested this against my own forked gitbook-cage and it looked good

File: ci/publish-book.sh [hunk 1]
```diff
@@ -59,13 +59,44 @@ echo --- create book repo
...
+    git push -f origin HEAD:master
+  else
+    echo "Content unchanged, publish skipped"
+  fi
+)
 
+echo --- update gitbook-cage
+(
+
+  if [[ -z $CI_BRANCH ]];
```
Comment (mvines) [inline ci/publish-book.sh]: nit: cuddle the `then`

**Comment by garious** [comment]
Thanks @rob-solana!

**Pull request merged**